### ISSUE TITLE
Warn when autoformatter clobbers files

### ISSUE THREAD

**Issue by JelleZijlstra** [Instagram/Fixit#208]
I'm sure I must have been doing something silly, but after running `python -m fixit.cli.apply_fix` with this config file:
```
allow_list_rules: []
block_list_patterns:
- '@generated'
- '@nolint'
block_list_rules: []
fixture_dir: ./fixtures
formatter: ["black"]
packages:
- fixit.rules
repo_root: .
rule_config: {}
use_noqa: false
```
Formatted files end up empty.

I was eventually able to resolve it by using `cat` as the formatter (and running black manually later). `black` (version 21.12b0) was installed in the venv I was running fixit in.

This was on proprietary code so I can't share all the details, but if anyone has any ideas for how this could happen, it would be great to improve Fixit so future users don't run into this case.

**Comment by zsol** [comment]
Try `formatter: ["black", "-"]` ;)

**Comment by JelleZijlstra** [comment]
Thanks, so I was indeed doing something silly :)

May I suggest that Fixit should detect if the formatter empties the file and tells the user?

**Comment by zsol** [comment]
Yeah I think that's a good improvement, but I'm not sure it would've helped in your case (you figured this out yourself)

**Comment by JelleZijlstra** [comment]
Well, at least it would have told me the issue was with the formatter and not with using relative paths or with the location of the config file or whatever else I thought of :)

And since presumably `black` is a very common formatter to use with Fixit, it would also be reasonable to warn specifically when we detect the formatter is set to `["black"]`.

**Comment by amyreese** [comment]
Hey folks, we just released Fixit 2.0, which includes major changes to the core frameworks, linting engine, API, and CLI. In light of this new release, this issue is likely to be resolved or no longer relevant. Please take a look at the new [user guide](https://fixit.readthedocs.io/en/latest/guide.html#) or the [upgrade guide](https://fixit.readthedocs.io/en/latest/upgrade.html) for more information on how to migrate your projects to the new version.

If you believe this issue is still relevant or still occurring in the latest version, please feel free to open a new issue, or reply here with more details.

Thank you!
